src/IpcEventDispatcher.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#include "IpcEventDispatcher.h"
+#include "IpcChannel.h"
+#include "PointViewerMainWindow.h"
+
+
+IpcEventDispatcher::IpcEventDispatcher(QObject* parent, PointViewerMainWindow* getPayload, 
+                                       QByteArray hookSpec, QByteArray hookPayload, int whichShortCut)
+    : QObject(parent),
```

Comment:
I made it a child of the main window at first, but thought this way Qt would take care of cleaning up during a gui session. The class is formatting the message and the instances keep the hook signals discernible, so IpcEventFormatter does make sense.